src/applications/personalization/profile360/components/ProfileView.jsx

```diff
@@ -200,9 +190,11 @@ class ProfileView extends React.Component {
 }
 
 function mapStateToProps(state) {
+  const directDepositIsAllowed = !directDepositIsBlocked(state);
   return {
     showDirectDepositLink:
-      directDepositIsSetUp(state) || directDepositAddressIsSetUp(state),
+      directDepositIsAllowed &&
```

Comment:
we're also adding this additional bit of logic to set `showDirectDepositLink` to `false` if `directDepositIsBlocked()` is `true`.

Reply:
Does this need the `isEligible` piece from the PaymentInformation `mapStateToProps`? If so, it would be cool if we could share that show/hide logic for `showDirectDepositLink ` and `shouldShowDirectDeposit` somehow. Not necessary to change either way though. This PR is busy enough 💫

Reply:
It doesn't need `isEligible` here. The PaymentInfo component is responsible for actually calling the endpoint that will in turn result in `directDepositIsSetUp()` or `directDepositAddressIsSetUp()` to return `true`. And that `isEligible` bool is really poorly named now. It used to make sense but now we have all sorts of new logic that really determines if a user is eligible for direct deposit. But this PR is not the time or place to fix that :)